addons/binding/org.openhab.binding.bluetooth.ruuvitag/src/main/java/org/openhab/binding/bluetooth/ruuvitag/handler/RuuviTagHandler.java

```diff
@@ -0,0 +1,163 @@
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.bluetooth.ruuvitag.handler;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.measure.Quantity;
+import javax.measure.Unit;
+import javax.measure.UnitConverter;
+import javax.measure.quantity.Acceleration;
+import javax.measure.quantity.Power;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.QuantityType;
+import org.eclipse.smarthome.core.library.unit.MetricPrefix;
+import org.eclipse.smarthome.core.library.unit.SIUnits;
+import org.eclipse.smarthome.core.library.unit.SmartHomeUnits;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.openhab.binding.bluetooth.BeaconBluetoothHandler;
+import org.openhab.binding.bluetooth.BluetoothDeviceListener;
+import org.openhab.binding.bluetooth.notification.BluetoothScanNotification;
+import org.openhab.binding.bluetooth.ruuvitag.RuuviTagBindingConstants;
+import org.slf4j.LoggerFactory;
+
+import fi.tkgwf.ruuvi.common.bean.RuuviMeasurement;
+import fi.tkgwf.ruuvi.common.parser.impl.AnyDataFormatParser;
+import tec.uom.se.function.LogConverter;
+import tec.uom.se.function.MultiplyConverter;
+import tec.uom.se.unit.TransformedUnit;
+
+/**
+ * The {@link RuuviTagHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Sami Salonen - Initial contribution
+ */
+@NonNullByDefault
+public class RuuviTagHandler extends BeaconBluetoothHandler implements BluetoothDeviceListener {
+
+    private static final double METRE_PER_SQUARE_SECOND_TO_STANDARD_GRAVITY = 9.80665;
+    private static final UnitConverter DIVIDE_BY_TEN = new MultiplyConverter(10).inverse();
+    private static final UnitConverter EXP_BASE_TEN = new LogConverter(10).inverse();
+
+    private static final Unit<Acceleration> STANDARD_GRAVITY = new TransformedUnit<>("g",
+            SmartHomeUnits.METRE_PER_SQUARE_SECOND, new MultiplyConverter(METRE_PER_SQUARE_SECOND_TO_STANDARD_GRAVITY));
+    private static final Unit<Power> DECIBEL_MILLIWATTS = new TransformedUnit<>("dBm",
+            MetricPrefix.MILLI(SmartHomeUnits.WATT), EXP_BASE_TEN.concatenate(DIVIDE_BY_TEN));
+
```

Comment:
You need to register the added quantity types to be able to use them:
```
static {
        SimpleUnitFormat.getInstance().label(STANDARD_GRAVITY, "g");
        SimpleUnitFormat.getInstance().label(DECIBEL_MILLIWATTS, "dBm");
    }
```
This requires an additional import `tec.uom.se.format` in `MANIFEST.MF`

Reply:
Is it a concern to have these defined in some other binding? What happens then?

Anyways, did as you suggested in ef8ca13.

Reply:
With #617 in place, should I remove these `label(...)` calls from this binding? I guess README needs some adjustments as well (standard gravity unit `gₙ` instead of `g`)

Reply:
Yes when https://github.com/openhab/openhab-core/pull/617 is in place you should remove them. The idea is to have all unit in openhab-core. Incidentally I needed to add some unit for another binding too, so I looked into it and see how this should be done. For that I added your units as a start. I did change the gravity unit as for 'standard' gravity that seems to be the right unit. What do you think?

There is a small catch. When these units are used from core it's not possible to use this binding with a 2.4 installation. I've had the same issue with other bindings and provided a jar under a relase tag on my own fork of openhab2-bindings and point people to that location (I also distribute the bindings through the eclipse market place).

Reply:
Thanks @Hilbrand. 

Now that https://github.com/openhab/openhab-core/pull/617 is merged, I removed all unit customization in this PR, please review d678427.

Reply:
Resolving this conversation.